hs_tools_resource/models.py

```diff
@@ -35,190 +33,112 @@ def metadata(self):
         md = ToolMetaData()
         return self._get_metadata(md)
 
-
 processor_for(ToolResource)(resource_processor)
 
 
 class RequestUrlBase(AbstractMetaDataElement):
-    term = 'Request Url Base'
-    value = models.CharField(null=True, max_length="500") # whatever the user gives us- format is "http://www.example.com/{resource-info}"
-
+    term = 'RequestUrlBase'
+    value = models.CharField(max_length=1024, null=True)
+    resShortID = models.CharField(max_length=128, default="UNKNOWN")
```

Comment:
@selimnairb @pkdash @ironfroggy @danames 

From my understanding, say, if we have a res obj, we can get its metadata object (res obj -> metadata obj), but we cannot do an inverse query (metadata obj X-> res obj). 

Our workflow for web app requires:  a regular res A obj (like raster) -> res A metadate obj -> webapp B metadata obj -> webappB  res obj

So the last arrow (webapp B metadata obj -> webappB  res obj) is our problem. As a workaround, currently we store the res id of a webapp res in the metadata obj of its own, and then we can use getResourceByID() to get the resource obj.

If there is a better way, we would like to change to apply it after AGU.